maven-core/src/main/java/org/apache/maven/execution/TransferListenerConfiguration.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.execution;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * The known modes of "transfer listener" logging.
+ *
+ * @since 4.0.0
+ */
+public final class TransferListenerConfiguration {
+
+    public enum Mode {
+        /**
+         * Silent, nothing logged that is transfer related.
+         */
+        QUIET,
+
+        /**
+         * Classic: as you knew Maven 3 since beginning.
+         */
+        CLASSIC,
+
+        /**
+         * Classic "light": similar to CLASSIC but without "Downloading..." lines (only "Downloaded...").
+         */
+        CLASSIC_LIGHT,
```

Comment:
I like this one which sounds like a good compromise. I have the suspicion that some of the complaints might come from users which use maven without a fully featured terminal which would allow in-place updates (e.g in CI). 

The classic mode would print every dl progress update into a new line which is not optimal. This is not a unique problem to maven, git does the exacts same last time I checked.